Added documentation describing our GDB support.

drwells opened the pull request:
We could probably advertise this better, but at least it is now mentioned in the manual (and also in the top level `readme`). I would like to hear if anyone has suggestion on how to improve upon this (maybe we should add `<blink>click here for GDB support</blink>` to every page in the manual to get the word out).

At some point in the near future I will beef up the wiki page a bit more (it is up to date, but very minimal right now).

bangerth commented on doc/users/gdb.html:
```diff
@@ -0,0 +1,57 @@
+  all <acronym>deal.II</acronym> source files):
+<pre class="sample">
+  cd dealii_source_directory
+  cp contrib/utilities/dotgdbinit.py ~/.gdbinit
```
This assumes that a user does not yet have such a file. Could we explain here (as well as the top of the dotgdbinit file) what to do if such a file already exists?

drwells commented:
I added a bit about configuring with multiple printers, though my description might require users to have some familiarity with Python. I also added more links.

bangerth commented:
Thanks!

bangerth commented:
This is the documentation part of #1578. It also addressess #902 .
